### ISSUE TITLE
Canu version is not being detected properly

### ISSUE THREAD

**Issue by tseemann** [sanger-pathogens/circlator#113]
Seems to be regular expressions with `v` etc?

```
% canu -version
Canu 1.6
# this is printed to stdout, not stderr
```

```
% circlator  progcheck
Circlator version: 1.5.1

External dependencies:
bwa	0.7.16	/usr/local/bin/bwa
Found canu but couldn't get version.
nucmer	3.1	/usr/local/bin/nucmer
prodigal	2.6.3	/usr/local/bin/prodigal
samtools	1.5	/usr/local/bin/samtools
WARNING: SPAdes version 3.11.0 is being used. It will work, but better results are usually obtained from Circlator using SPAdes version 3.7.1. Although 3.7.1 is not the latest version, we recommend it for Circlator.
spades	3.11.0	/usr/local/bin/spades.py

Python version:
3.6.2 (default, Jul 17 2017, 16:44:45)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]

Python dependencies:
openpyxl	2.4.1	/usr/local/Cellar/circlator/1.5.1/libexec/lib/python3.6/site-packages/openpyxl/__init__.py
pyfastaq	3.14.0	/usr/local/Cellar/circlator/1.5.1/libexec/lib/python3.6/site-packages/pyfastaq/__init__.py
pymummer	0.10.1	/usr/local/Cellar/circlator/1.5.1/libexec/lib/python3.6/site-packages/pymummer/__init__.py
pysam	0.8.4	/usr/local/Cellar/circlator/1.5.1/libexec/lib/python3.6/site-packages/pysam/__init__.py
```

Code regexp:
```
'canu': ('-version', re.compile('^Canu \D*([\d][\d\.]+)')),
```

Debug:
```
Running this command to get version: canu -version
__________ (begin output from canu -version)___________
Canu 1.6
__________ (end of output from canu -version)___________
Looking far a match to the regex "^Canu.*v.?([0-9][0-9\.]+)" in the above output
No match found to the regex
Found canu but couldn't get version.
```

```
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/circlator/1.5.1_1/bin/circlator", line 58, in <module>
    exec('circlator.tasks.' + task + '.run()')
  File "<string>", line 1, in <module>
  File "/home/linuxbrew/.linuxbrew/Cellar/circlator/1.5.1_1/libexec/lib/python3.6/site-packages/circlator/tasks/all.py", line 72, in run
    circlator.versions.get_all_versions(sys.stdout, raise_error=True, assembler=options.assembler)
  File "/home/linuxbrew/.linuxbrew/Cellar/circlator/1.5.1_1/libexec/lib/python3.6/site-packages/circlator/versions.py", line 15, in get_all_versions
    external_progs.check_all_progs(verbose=False, raise_error=raise_error, filehandle=filehandle, debug=debug, assembler=assembler)
  File "/home/linuxbrew/.linuxbrew/Cellar/circlator/1.5.1_1/libexec/lib/python3.6/site-packages/circlator/external_progs.py", line 108, in check_all_progs
    make_and_check_prog(prog, verbose=verbose, raise_error=raise_error, filehandle=filehandle, debug=debug, required=prog==assembler)
  File "/home/linuxbrew/.linuxbrew/Cellar/circlator/1.5.1_1/libexec/lib/python3.6/site-packages/circlator/external_progs.py", line 81, in make_and_check_prog
    handle_error('Found ' + name + " but couldn't get version.", raise_error=raise_error)
  File "/home/linuxbrew/.linuxbrew/Cellar/circlator/1.5.1_1/libexec/lib/python3.6/site-packages/circlator/external_progs.py", line 56, in handle_error
    raise Error(message + '\nCannot continue')
circlator.external_progs.Error: Found canu but couldn't get version.
Cannot continue
```

**Comment by jnesme** [comment]
Fixed by installing canu 1.6 from source instead of Home/Linux-brew recipe.
```
Circlator version: 1.5.1

External dependencies:
bwa     0.7.16  /usr/local/home/jnesme/.linuxbrew/bin/bwa
canu    1.6     /mibi/jnesme/Bin/canu/Linux-amd64/bin/canu
nucmer  3.1     /usr/local/home/jnesme/.linuxbrew/bin/nucmer
prodigal        2.6.3   /usr/local/home/jnesme/.linuxbrew/bin/prodigal
samtools        1.5     /usr/local/home/jnesme/.linuxbrew/bin/samtools
spades  3.7.1   /mibi/jnesme/Bin/SPAdes-3.7.1-Linux/bin/spades.py

Python version:
3.4.2 (default, Oct  8 2014, 10:45:20)
[GCC 4.9.1]

Python dependencies:
openpyxl        2.4.8   /usr/local/home/jnesme/.local/lib/python3.4/site-packages/openpyxl/__init__.py
pyfastaq        3.15.0  /usr/local/home/jnesme/.local/lib/python3.4/site-packages/pyfastaq/__init__.py
pymummer        0.10.2  /usr/local/home/jnesme/.local/lib/python3.4/site-packages/pymummer-0.10.2-py3.4.egg/pymummer/__init__.py
pysam   0.11.2.2        /usr/local/home/jnesme/.local/lib/python3.4/site-packages/pysam/__init__.py
```

**Comment by tillea** [comment]
Hi,
I've fixed this in the Debian package with this [patch](https://anonscm.debian.org/git/debian-med/circlator.git/tree/debian/patches/fix_canu_detection.patch).
Kind regards, Andreas.

**Comment by andrewjpage** [comment]
This was fixed a little while ago. I've added a test to check for this version string in canu.

**Comment by SabineSN** [comment]
Hi,
I have the same problem with circlator, where doesn't find canu version (I have version 1.9). I have tried reinstalling both circlator and canu. I have also tried installing older canu versions (1.4 and 1.6) and I have tried this change: https://salsa.debian.org/med-team/circlator/blob/master/debian/patches/fix_canu_detection.patch.
Nothing is working, circlator still gives this output:

Circlator version: 1.5.5

External dependencies:
bwa	0.7.17	/home/kma/miniconda3/bin/bwa
Found canu but couldn't get version.
nucmer	3.1	/home/kma/miniconda3/bin/nucmer
prodigal	2.6.3	/home/kma/miniconda3/bin/prodigal
samtools	1.7	/usr/bin/samtools
WARNING: SPAdes version 3.14.0 is being used. It will work, but better results are usually obtained from Circlator using SPAdes version 3.7.1. Although 3.7.1 is not the latest version, we recommend it for Circlator.
spades	3.14.0	/home/kma/miniconda3/bin/spades.py

Does anyone have ideas to solving this problem?
Kind regards, Sabine
